typescript/schedules.md

```diff
@@ -0,0 +1,1255 @@
+
+  /**
+   * Trigger one Action immediately.
+   *
+   * @default false
+   */
+  triggerImmediately?: boolean;
+
+  /**
+   * Runs though the specified time periods and takes Actions as if that time passed by right now, all at once. The
+   * overlap policy can be overridden for the scope of the backfill.
+   */
+  backfill?: Backfill[];
+
+  /**
+   * Additional non-indexed information attached to the Schedule. The values can be anything that is
+   * serializable by the {@link DataConverter}.
+   */
+  memo?: Record<string, any>;
+
+  /**
+   * Additional indexed information attached to the Schedule. More info:
+   * https://docs.temporal.io/docs/typescript/search-attributes
+   *
+   * Values are always converted using {@link JsonPayloadConverter}, even when a custom Data Converter is provided.
+   */
+  searchAttributes?: SearchAttributes;
+}
+
+export interface ListScheduleOptions {
+  /**
+   * How many results to return
+   * @default 1000
+   */
+  pageSize?: number;
+
+  /** Token to get the next page of results */
+  nextPageToken?: Base64;
+}
+
+/**
+ * Client for starting Workflow executions and creating Workflow handles
+ */
+export class ScheduleClient {
+  public readonly options: LoadedScheduleClientOptions;
+  public readonly connection: ConnectionLike;
+
+  constructor(options?: ScheduleClientOptions) {
+    this.connection = options?.connection ?? Connection.lazy();
+    this.options = {
+      ...defaultScheduleClientOptions(),
+      ...options,
+      loadedDataConverter: loadDataConverter(options?.dataConverter),
+    };
+  }
+
+  /**
+   * Raw gRPC access to the Temporal service. Schedule-related methods are included in {@link WorkflowService}.
+   *
+   * **NOTE**: The namespace provided in {@link options} is **not** automatically set on requests made to the service.
+   */
+  get scheduleService(): WorkflowService {
+    return this.connection.workflowService;
+  }
+
+  /**
+   * Set the deadline for any service requests executed in `fn`'s scope.
+   */
+  async withDeadline<R>(deadline: number | Date, fn: () => Promise<R>): Promise<R> {
+    return await this.connection.withDeadline(deadline, fn);
+  }
+
+  /**
+   * Set metadata for any service requests executed in `fn`'s scope.
+   *
+   * @returns returned value of `fn`
+   *
+   * @see {@link Connection.withMetadata}
+   */
+  async withMetadata<R>(metadata: Metadata, fn: () => Promise<R>): Promise<R> {
+    return await this.connection.withMetadata(metadata, fn);
+  }
+
+  /**
+   * Create a new Schedule.
+   */
+  public async create<Action extends ScheduleActionType>(options: ScheduleOptions<Action>): Promise<ScheduleHandle> {  }
+
+  /**
+   * List Schedules with an `AsyncIterator`:
+   *
+   * ```ts
+   * for await (const schedule: Schedule of client.list()) {
+   *   const { id, memo, searchAttributes } = schedule
+   *   // ...
+   * }
+   * ```
+   */
+  public list(options?: ListScheduleOptions): AsyncIterator<Schedule> {}
```

Comment:
You should consider adding high-level list for workflows if you're gonna add it for schedules

Reply:
We'll add a list workflows API to all SDKs, it's requested frequently.